### PR TITLE
fix: Errorbar breaking at category XAxis type!

### PR DESCRIPTION
**Pull request by AnujSharma141**
## Description

Added invariant that checks for x direction and category type.

## Related Issue

Closes [#3812 ]

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [] I have added a storybook story or extended an existing story to show my changes
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/cartesian/ErrorBar.tsx [hunk 1]
```diff
@@ -49,6 +50,9 @@ export type Props = SVGProps<SVGLineElement> & ErrorBarProps;
 export function ErrorBar(props: Props) {
   const { offset, layout, width, dataKey, data, dataPointFormatter, xAxis, yAxis, ...others } = props;
   const svgProps = filterProps(others);
+
+  invariant(!(props.direction === 'x' && xAxis.type !== 'number'), 'Errorbar requires Axis type to be "number".');
```
Comment (ckifer) [inline src/cartesian/ErrorBar.tsx:54]: can you make the text `ErrorBar` to match the component casing?

**Review by ckifer — CHANGES_REQUESTED** [review]
One small thing.

Also, did you make sure to run and check that this gets thrown when non-number types are provided?

**Comment by ckifer** [comment]
Would you consider making a small update to the docs too?

File: src/cartesian/ErrorBar.tsx (hunk 1 above)
Comment (AnujSharma141) [reply src/cartesian/ErrorBar.tsx:54]: sure will do this, also I did test the variant manually it throws when ErrorBar direction is x but a non number type is set to XAxis.

**Comment by ckifer** [comment]
Over in the recharts.org repo we could make the same documentation change, but looks good for now. Thank you!

**Pull request merged**